bindings/java/src/junit/com/apple/foundationdb/EventKeeperTest.java

```diff
@@ -0,0 +1,145 @@
+			long jniCalls = timer.getCount(Events.JNI_CALL);
+
+			Assertions.assertEquals(1L, jniCalls, "Unexpected number of JNI calls:");
+		} catch (UnsatisfiedLinkError ignored) {
+		}
+	}
+
+	@Test
+	void testGetReadVersion() throws Exception {
+		EventKeeper timer = new TestKeeper();
+
+		try (FDBTransaction txn = new FDBTransaction(1, null, null, timer)) {
+			Assertions.assertThrows(UnsatisfiedLinkError.class,
+			                        () -> { txn.getReadVersion(); }, "Test should call a bad native method");
+			long jniCalls = timer.getCount(Events.JNI_CALL);
+
+			Assertions.assertEquals(1L, jniCalls, "Unexpected number of JNI calls:");
+		} catch (UnsatisfiedLinkError ignored) {
+		}
+	}
+
+	@Test
+	void testGetRangeRecordsFetches() throws Exception {
+		EventKeeper timer = new TestKeeper();
+		List<KeyValue> testKvs = Arrays.asList(new KeyValue("hello".getBytes(), "goodbye".getBytes()));
+
+		FDBTransaction txn = new FakeFDBTransaction(testKvs, 1L, null, null);
+
+		RangeQuery query = new RangeQuery(txn, true, KeySelector.firstGreaterOrEqual(new byte[] { 0x00 }),
+		                                  KeySelector.firstGreaterOrEqual(new byte[] { (byte)0xFF }), -1, false,
+		                                  StreamingMode.ITERATOR, timer);
+		AsyncIterator<KeyValue> iter = query.iterator();
+
+		List<KeyValue> iteratedItems = new ArrayList<>();
+		while (iter.hasNext()) {
+			iteratedItems.add(iter.next());
+		}
+
+		// basic verification that we got back what we expected to get back.
+		Assertions.assertEquals(testKvs.size(), iteratedItems.size(), "Incorrect iterated list, size incorrect.");
+
+		int expectedByteSize = 0;
+		for (KeyValue expected : testKvs) {
+			byte[] eKey = expected.getKey();
+			byte[] eVal = expected.getValue();
+			expectedByteSize += eKey.length + 4;
+			expectedByteSize += eVal.length + 4;
+			boolean found = false;
+			for (KeyValue actual : iteratedItems) {
+				byte[] aKey = actual.getKey();
+				byte[] aVal = actual.getValue();
+				if (ByteArrayUtil.compareTo(eKey, 0, eKey.length, aKey, 0, aKey.length) == 0) {
+					int cmp = ByteArrayUtil.compareTo(eVal, 0, eVal.length, aVal, 0, aVal.length);
+					Assertions.assertEquals(0, cmp, "Incorrect value returned");
+					found = true;
+					break;
+				}
+			}
+
+			Assertions.assertTrue(found, "missing key!");
+		}
+
+		// now check the timer and see if it recorded any events
+		Assertions.assertEquals(1, timer.getCount(Events.RANGE_QUERY_FETCHES), "Unexpected number of chunk fetches");
+		Assertions.assertEquals(testKvs.size(), timer.getCount(Events.RANGE_QUERY_RECORDS_FETCHED),
+		                        "Unexpected number of tuples fetched");
+		Assertions.assertEquals(expectedByteSize, timer.getCount(Events.BYTES_FETCHED),
+		                        "Incorrect number of bytes fetched");
+	}
+
+	/* private helper methods and classes */
+	private static class TestKeeper implements EventKeeper {
```

Comment:
Should this be replaced with a `MapEventKeeper` now that that's a thing?

Reply:
sure